src/isl.rs

```diff
@@ -201,11 +218,79 @@ impl IslTypeRef {
                 }
             }
             IslTypeRef::AnonymousType(isl_type) => {
+                let type_id = context.borrow().get_total_types();
                 let type_def = TypeDefinition::parse_from_isl_type_and_update_type_store(
-                    isl_type, type_store,
+                    isl_type, type_store, context,
                 )?;
-                Ok(type_store.borrow_mut().add_anonymous_type(type_def))
+                // get the last added anonymous type's type_id for given anonymous type
+                Ok(type_id)
             } //TODO: add a check for ImportType type reference here
         }
     }
 }
+
+#[cfg(test)]
+mod isl_tests {
+    use super::*;
+    use ion_rs::value::owned::{text_token, OwnedElement, OwnedStruct};
+    use ion_rs::value::Builder;
+    use rstest::*;
+    use std::iter::FromIterator;
+
+    #[rstest(
+    isl_type1,isl_type2,
+    case::type_constraint_with_anonymous_type(
+        /* For a schema with single anonymous type as below: 
+            { type: int }
+         */
+        (&OwnedStruct::from_iter(vec![("type", OwnedElement::from(text_token("int")))].into_iter())).try_into().unwrap(),
+        IslType::new(None, vec![IslConstraint::Type(IslTypeRef::CoreIslType(IonType::Integer))])
+    ),
```

Comment:
These `OwnedElement` construction snippets get pretty dense. I suggest using `ElementReader` and writing out the Ion text instead.

```rust
use crate::value::reader::{element_reader, ElementReader};
// ...

// new helper function
fn load(text: &str) -> OwnedElement {
    element_reader()
        .read_one(text.as_bytes())
        .expect("parsing failed unexpectedly")
}

// ...
case::type_constraint_with_anonymous_type(
    load(r#" // Raw string, no escaping of special characters required
        {type: int}
    "#),
    IslType::new(None, vec![IslConstraint::Type(IslTypeRef::CoreIslType(IonType::Integer))])
),
```